src/modules/bppp/bppp_norm_product_impl.h

```diff
@@ -238,17 +242,22 @@ static int secp256k1_bppp_rangeproof_norm_product_prove(
     ecmult_r_cb_data r_cb_data;
     size_t g_len = n_vec_len, h_len = l_vec_len;
     const size_t G_GENS_LEN = g_len;
-    size_t log_g_len, log_h_len;
-    size_t num_rounds;
 
-    VERIFY_CHECK(g_len > 0 && h_len > 0);
-    log_g_len = secp256k1_bppp_log2(g_len);
-    log_h_len = secp256k1_bppp_log2(h_len);
-    num_rounds = log_g_len > log_h_len ? log_g_len : log_h_len;
-    /* Check proof sizes.*/
-    VERIFY_CHECK(*proof_len >= 65 * num_rounds + 64);
-    VERIFY_CHECK(g_vec_len == (n_vec_len + l_vec_len) && l_vec_len == c_vec_len);
-    VERIFY_CHECK(secp256k1_is_power_of_two(n_vec_len) && secp256k1_is_power_of_two(c_vec_len));
+#ifdef VERIFY
+    {
+        size_t log_g_len_ver = secp256k1_bppp_log2(g_len);
+        size_t log_h_len_ver = secp256k1_bppp_log2(h_len);
+        size_t num_rounds_ver = log_g_len_ver > log_h_len_ver ? log_g_len_ver : log_h_len_ver;
+        VERIFY_CHECK(g_len > 0 && h_len > 0);
```

Comment:
It would be better to VERIFY_CHECK g_len and h_len before log2, because the function doc says that one must not call log2 on input 0. I think the reason is not only that the result would be wrong but also that the result of `__builtin_clz` is "undefined" for 0.